libraries/AP_ExternalAHRS/AP_ExternalAHRS_VectorNav.h

```diff
@@ -63,20 +63,23 @@ class AP_ExternalAHRS_VectorNav : public AP_ExternalAHRS_backend {
 
     void initialize();
 
-    void process_ins_packet1(const uint8_t *b);
-    void process_ins_packet2(const uint8_t *b);
-    void process_ahrs_packet(const uint8_t *b);
     void run_command(const char *fmt, ...);
+    void process_imu_packet(const uint8_t *b);
+    void process_ahrs_ekf_packet(const uint8_t *b);
+    void process_ins_ekf_packet(const uint8_t *b);
+    void process_ins_gnss_packet(const uint8_t *b);
 
     uint8_t *pktbuf;
     uint16_t pktoffset;
     uint16_t bufsize;
 
-    struct VN_INS_packet1 *last_ins_pkt1;
-    struct VN_INS_packet2 *last_ins_pkt2;
+    struct VN_imu_packet const *latest_imu_packet = nullptr;
+    struct VN_INS_ekf_packet const *latest_ins_ekf_packet = nullptr;
+    struct VN_INS_gnss_packet const *latest_ins_gnss_packet = nullptr;
 
-    uint32_t last_pkt1_ms;
-    uint32_t last_pkt2_ms;
+    uint32_t last_pkt1_ms = UINT32_MAX;
+    uint32_t last_pkt2_ms = UINT32_MAX;
+    uint32_t last_pkt3_ms = UINT32_MAX;
```

Comment:
Why not 0?

Reply:
Agreed, will update

Reply:
If zero you don't need to initialize manually (you get it for frees link nullptr).